### PR TITLE
[Narwhal] remove external consensus from primary

### PR DESCRIPTION
**Pull request by mwtian**
## Description 

External consensus has not been updated or actively used for awhile as far as we know. Removing them can simplify the Narwhal codebase and make some future changes easier.

#12946

## Test Plan 

CI
Narwhal nightly: https://github.com/MystenLabs/sui/actions/runs/5522777741/jobs/10072838309

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: narwhal/node/src/main.rs [hunk 1]
```diff
@@ -71,10 +71,6 @@ async fn main() -> Result<(), eyre::Report> {
                 .args_from_usage("--workers=<FILE> 'The file containing worker information'")
                 .args_from_usage("--parameters=[FILE] 'The file containing the node parameters'")
                 .args_from_usage("--store=<PATH> 'The path where to create the data store'")
-                .subcommand(SubCommand::with_name("primary")
-                    .about("Run a single primary")
```
Comment (akichidis) [inline narwhal/node/src/main.rs:75]: I believe we need this so we can run a `primary` node - the arg is indeed not needed

File: narwhal/primary/src/primary.rs [hunk 2]
```diff
@@ -157,11 +147,6 @@ impl Primary {
             &primary_channel_metrics.tx_certificate_fetcher,
             &primary_channel_metrics.tx_certificate_fetcher_total,
         );
-        let (tx_block_synchronizer_commands, rx_block_synchronizer_commands) = channel_with_total(
-            CHANNEL_CAPACITY,
-            &primary_channel_metrics.tx_block_synchronizer_commands,
```
Comment (akichidis) [inline narwhal/primary/src/primary.rs:162]: Let's also remove the metrics from here https://github.com/MystenLabs/sui/blob/c767d31e180ee2ebb5e722a9ea821ad374a21716/narwhal/primary/src/metrics.rs#L74C9-L74C39

**Review by akichidis — APPROVED** [review]
Red is good!! Thanks @mwtian . Also are we planning to remove the additional stuff from https://github.com/MystenLabs/sui/tree/main/narwhal/dag and https://github.com/MystenLabs/sui/blob/main/narwhal/consensus/src/dag.rs on separate PR?

**Review by arun-koshy — APPROVED** [review]

File: narwhal/node/src/main.rs (hunk 1 above)
Comment (mwtian) [reply narwhal/node/src/main.rs:75]: Reverted.

**Comment by mwtian** [comment]
> External consensus has not been updated or actively used for awhile as far as we know. Removing them can simplify the Narwhal codebase and make some future changes easier.

Yes DAGs will be removed in another PR.

**Pull request merged**